src/Components/Assets/AssetServiceEditModal.tsx

```diff
@@ -169,19 +168,32 @@ export const AssetServiceEditModal = (props: {
               className="col-span-6 sm:col-span-3"
               data-testid="asset-last-serviced-on-input"
             >
-              <FieldLabel>Serviced On</FieldLabel>
-              <DateInputV2
+              <DateFormField
+                // eslint-disable-next-line i18next/no-literal-string
+                label="Serviced On"
                 name="serviced_on"
                 className="mt-2"
                 position="LEFT"
                 value={new Date(form.serviced_on)}
                 onChange={(date) => {
-                  setForm({
-                    ...form,
-                    serviced_on: dayjs(date).format("YYYY-MM-DD"),
-                  });
+                  if (
+                    dayjs(date.value).format("YYYY-MM-DD") >
+                    new Date(
+                      props.service_record.created_date
+                    ).toLocaleDateString("en-ca")
+                  ) {
+                    Notification.Error({
+                      msg: `Service date can't be after ${dayjs(
+                        props.service_record.created_date
+                      ).format("DD/MM/YYYY")} (Creation date)`,
+                    });
```

Comment:
you may use `formatDate` utility instead of manually specifying the format here. This way we can maintain the date format from a single point if we ever need to change it.